portfolio/src/main/java/com/google/sps/servlets/ChartServlet.java

```diff
@@ -41,40 +43,29 @@
 import javax.servlet.http.HttpServletResponse;
 
 /* Servlet to read chart data from data store and return it in doGet */
-// TODO(briafassler): Fix the problem where chart only appears after pressing submit
 @WebServlet("/chart-data")
 public class ChartServlet extends HttpServlet {
   private static final String TABLE_NAME = "Police Killings By Year";
-  private static final String YEAR = "year";
-  private static final String AMOUNT_KILLED = "amount killed";
-  private static final String test = "{'2020': '5'}";
+  private static final String DATE = "Date";
+  private static final String TOTALS = "Totals";
+  private static final String YEAR_ORDER = "Years";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query(TABLE_NAME);
+    Query query = new Query(TABLE_NAME).addSort(DATE, SortDirection.ASCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     List<DataPair> dataPairs = new ArrayList<DataPair>();
     for (Entity entity : results.asIterable()) {
-      int amountKilled = (int) entity.getProperty(AMOUNT_KILLED);
-      String year = (String) entity.getProperty(YEAR);
+      int amountKilled = ((Long) entity.getProperty(TOTALS)).intValue();
+      int year = ((Long) entity.getProperty(DATE)).intValue();
```

Comment:
interesting casting method

Reply:
gotta give credit to kenneth, he helped me out